### PR TITLE
Add Email Notification On Registration

### PR DESCRIPTION
**Pull request by Aayushchy**
An email will be sent when user is registered successfully

### CODE REVIEW

File: src/main/java/com/esewa/usermanagement/configuration/EmailConfigProperties.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Getter
+@Setter
+@Configuration
+@ConfigurationProperties(prefix = "email")
+public class EmailConfigProperties {
```
Comment (devintern-esewa) [inline src/main/java/com/esewa/usermanagement/configuration/EmailConfigProperties.java:12]: Can you please create a table to hold these templates?

File: src/main/java/com/esewa/usermanagement/service/EmailServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        this.javaMailSender = javaMailSender;
+        this.emailConfigProperties = emailConfigProperties;
+    }
+
+    @Async
+    @Override
+    public void sendEmail(String recipientEmail) {
+
+        try {
+            SimpleMailMessage mailMessage = new SimpleMailMessage();
```
Comment (devintern-esewa) [inline src/main/java/com/esewa/usermanagement/service/EmailServiceImpl.java:30]: Can you also log these emails in email_history table?

**Review by devintern-esewa — CHANGES_REQUESTED** [review]
Verified

File: src/main/java/com/esewa/usermanagement/configuration/EmailConfigProperties.java (hunk 1 above)
Comment (Aayushchy) [reply src/main/java/com/esewa/usermanagement/configuration/EmailConfigProperties.java:12]: Done

File: src/main/java/com/esewa/usermanagement/service/EmailServiceImpl.java (hunk 2 above)
Comment (Aayushchy) [reply src/main/java/com/esewa/usermanagement/service/EmailServiceImpl.java:30]: Done

**Review by devintern-esewa — APPROVED** [review]

**Pull request merged**